packages/viewport/src/with-viewport-match.js

```diff
@@ -32,13 +31,33 @@ import { withSelect } from '@wordpress/data';
  *
  * @return {Function} Higher-order component.
  */
-const withViewportMatch = ( queries ) => createHigherOrderComponent(
-	withSelect( ( select ) => {
-		return mapValues( queries, ( query ) => {
-			return select( 'core/viewport' ).isViewportMatch( query );
-		} );
-	} ),
-	'withViewportMatch'
-);
+const withViewportMatch = ( queries ) => {
+	const useViewPortQueriesResult = () => mapValues( queries, ( query ) => {
+		let [ operator, breakpointName ] = query.split( ' ' );
+		if ( breakpointName === undefined ) {
+			breakpointName = operator;
+			operator = '>=';
+		}
+		// Hooks should unconditionally execute in the same order,
+		// we are respecting that as from the static query of the HOC we generate
+		// a hook that calls other hooks always in the same order (because the query never changes).
+		// eslint-disable-next-line react-hooks/rules-of-hooks
+		return useViewportMatch( breakpointName, operator );
+	} );
+	return createHigherOrderComponent(
+		( WrappedComponent ) => {
+			return ( props ) => {
+				const queriesResult = useViewPortQueriesResult();
+				return (
+					<WrappedComponent
+						{ ...props }
+						{ ...queriesResult }
```

Comment:
`withSelect` served as a memoization layer which this code doesn't replicate.

`queriesResult` will change on every render here. Could it be a performance concern?

Reply:
Hi @epiqueras, I don't think it is a concern, we are spreading queries result and passing each prop individually as a boolean e.g: isLarge: false, isSmall: true etc... Even if withViewportMatch is used in a pure component as we are just passing boolean values if useViewportMatch did not change we will not trigger unnecessary rerenders.

Reply:
Right, but `withSelect` wraps using a pure component, this HOC does not.

Reply:
Good point, I kept the previous behavior and also used pure HOC as the withSelect does.

Reply:
Nice.